drivers/analog/Kconfig

```diff
@@ -17,6 +17,7 @@ if ANALOG
 config ADC
 	bool "Analog-to-Digital Conversion"
 	default n
+	depends on ALLOW_BSD_COMPONENTS
```

Comment:
@jerpelea I think those file with Copyright from other people that we didn't contact need to be reviewed. In the case of adc.c for example if you run "git blame" you see that "Li Zhuoyi" contributions to the file were removed, but his copyright still there. It is an error and need to be fixed. Just putting it as BSD component will make the already complex config even more complex. How on earth people will image that to use ADC on NuttX he needs to enable BSD license?

Reply:
@acassis during the last months I contacted developer and waited for them to answer. For the files that we can clear the license I would gladly move them to Apache instead of adding BSD barrier.